### PR TITLE
Show buttons and title when window is dragged to top panel.

### LINKED ISSUE #48: Title in activity bar  not updated when fullscreen-ing with mouse
If I fullscreen a window with the Super-Up shortcut or by double clicking the title bar, the activity bar is properly updated (showing the full window title).
But if I fullscreen a window by dragging its title bar to the top, the activity bar still shows the default application name.

It looks like it could be a bug in GNOME (version 3.18.2 from Fedora 23). `window.get_maximized()` in Util.getWindow() does think that the upper window is not maximized at this point. I guess it means that the signal is sent too early when the fullscreen is made by dragging with the mouse.
But I'll let you diagnose (in case I missed something) and decide if that is to be reported upstream (if you do, could you post the GNOME bug report here too please?).

Fullscreen by dragging to the top of the screen with mouse (wrong):
screenshot from 2015-12-12 15-34-49

Fullscreen by Super+Up or double clicking on the title bar (correct):
screenshot from 2015-12-12 15-34-55

Thanks.

### LINKED ISSUE #81: Window title and buttons are not showing on the top panel when dragging window on it.
When dragging window on top panel to maximize it, titlebar is hiding properly but window title and buttons are not showing on panel. I will provide pull-request to fix it.

### PR DESCRIPTION
**Pull request by polter-rnd**
Added showing buttons and overriding title when dragging window on top panel.
Fixes #48.

### CODE REVIEW

File: pixel-saver@deadalnix.me/app_menu.js [hunk 1]
```diff
@@ -197,10 +197,12 @@ function enable() {
 		wmCallbackIDs.push(global.window_manager.connect('maximize', updateAppMenu));
 		wmCallbackIDs.push(global.window_manager.connect('unmaximize', updateAppMenu));
 	} catch (e) {
-		// Gnome 3.18
+		// Gnome 3.18+
 		wmCallbackIDs.push(global.window_manager.connect('size-change', updateAppMenu));
 	}
 	
+	wmCallbackIDs.push(global.window_manager.connect('hide-tile-preview', updateAppMenu));
```
Comment (deadalnix) [inline pixel-saver@deadalnix.me/app_menu.js]: Does this work on 3.16 ?

File: pixel-saver@deadalnix.me/buttons.js [hunk 2]
```diff
@@ -247,10 +247,13 @@ function enable() {
 		wmCallbackIDs.push(global.window_manager.connect('maximize', updateVisibility));
 		wmCallbackIDs.push(global.window_manager.connect('unmaximize', updateVisibility));
 	} catch (e) {
-		// Gnome 3.18
+		// Gnome 3.18+
 		wmCallbackIDs.push(global.window_manager.connect('size-change', updateVisibility));
 	}
 	
+	// Needed for showing buttons on window drag to top panel
+	wmCallbackIDs.push(global.window_manager.connect('hide-tile-preview', updateVisibility));
```
Comment (deadalnix) [inline pixel-saver@deadalnix.me/buttons.js]: dito

**Comment by satriani-vai** [comment]
I have noticed another one issue
If the window dragged to left or right site, then the title bar disappears also the buttons is not shown.
I think it is useful to disable this feature for this specific case.
bildschirmfoto von 2016-07-11 12-09-25

**Comment by deadalnix** [comment]
For the left/right maximized, it looks liek gtk changed it behavior in that regard. The extention is not the one undecorating the window.

@polter-rnd I tried to google for hide-tile-preview but couldn't find anything. This is a useful bugfix, could you help me understand it so i can merge ?

**Comment by polter-rnd** [comment]
@deadalnix hide-tile-preview was introduced in this commit: https://mail.gnome.org/archives/commits-list/2014-February/msg07406.html (available since version 3.12).

In gnome-shell, when user drags window to screen side (left, right, or top), tile preview is shown (rectangle showing how window will be layed out after dropping). So hide-tile-preview is emitted when user drops window on any side, because tile preview hides at this moment.

We can use this behaviour to determine that user maximized window by dragging (since there are no other signals available for that).

**Comment by deadalnix** [comment]
Ok got you. Thanks.

**Pull request merged**